tests/testthat/test-schedule.R

```diff
@@ -0,0 +1,87 @@
+test_that("get_timezones() gets timeszones from v1 url when available", {
+  # With version available
+  client <- MockConnect$new()
+
+  client$mock_response(
+    path = "v1/timezones",
+    content = list(
+      list(timezone = "Africa/Abidjan", offset = "+00:00"),
+      list(timezone = "Africa/Accra", offset = "+00:00")
+    )
+  )
+  client$mock_response(
+    path = "timezones",
+    content = list(
+      list(timezone = "Africa/Abidjan", offset = "+00:00"),
+      list(timezone = "Africa/Accra", offset = "+00:00")
+    )
+  )
+
+  expect_equal(
+    get_timezones(client),
+    list(`Africa/Abidjan (+00:00)` = "Africa/Abidjan", `Africa/Accra (+00:00)` = "Africa/Accra")
+  )
+  expect_equal(
+    client$call_log,
+    "GET https://connect.example/__api__/v1/timezones"
+  )
+})
+
+test_that("get_timezones() gets timeszones from unversioned url when v1 returns 404", {
+  client <- MockConnect$new()
+
+  client$mock_response(
+    path = "v1/timezones",
+    content = "404 page not found",
+    status_code = 404L,
+    headers = c("Content-Type" = "text/plain; charset=utf-8")
+  )
```

Comment:
I personally like this abstraction quite a bit. It removes a lot of the boilerplate and clarifies what's being augmented for the purpose of the test. 

Also, This looks fairly similar to the `responses` package we are using in Python.